### ISSUE TITLE
Cannot reproduce results

### ISSUE THREAD

**Issue by TicTacToePP** [zcxcf/EA-ViT#2]
Dear authors @zcxcf ,

I have tried your code on CIFAR 10 and the results for stage 2 models shows like this:

STAGE='inference'
BATCH_SIZE=16
DATASET="cifar10_full"
DEVICE=cuda:1
CHECKPOINT="./logs_weight/stage_2/cifar10_full/Oct03_09-40-29/weight/stage2.pth"
CONSTRAINT=0.5

python inference.py \
  --stage $STAGE \
  --batch_size $BATCH_SIZE \
  --dataset $DATASET \
  --stage2_checkpoint_path $CHECKPOINT \
  --constraint $CONSTRAINT \
  --device $DEVICE \

start eval
macs 2.39 GMACs
eval: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▌| 623/625 [00:35<00:00, 17.64it/s<class 'dict'>]
val acc 16.45
eval: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 625/625 [00:35<00:00, 17.43it/s<class 'dict'>]

I wonder if you could help with this. Thanks.

**Comment by zcxcf** [comment]
From your experimental results, it seems that the model hasn’t been fine-tuned on the CIFAR-10 dataset yet (the accuracy is very low, close to the 10% random baseline). The weights provided in my README are re-arranged pretrained weights, which need to be fine-tuned on downstream datasets first.

After fine-tuning, you can set the constraint to 1 to check whether the largest submodel reaches the expected accuracy before evaluating the performance of other submodels.

**Comment by TicTacToePP** [comment]
@zcxcf Thank you for the response. I download the rearranged_para.pth first. My experiment order is stage1 training - search submodel - stage2 training, I ran the first stage training with:

`#!/bin/bash
STAGE='stage1'
BATCH_SIZE=256
DATASET="cifar10_full"
DEVICE="cuda:0"
MAX_LR=1e-5
MIN_LR=1e-7
CHECKPOINT="./experiments/rearranged_para.pth"

python train_stage1.py \
  --stage $STAGE \
  --batch_size $BATCH_SIZE \
  --max_lr $MAX_LR \
  --min_lr $MIN_LR \
  --dataset $DATASET \
  --rearranged_checkpoint_path $CHECKPOINT \
  --device $DEVICE \`

search submodel:

`#!/bin/bash

STAGE='stage2_nsga'
BATCH_SIZE=256
DATASET="cifar10_full"
DEVICE="cuda:0"
CHECKPOINT="./experiments/logs_weight/stage_1/cifar10_full/Oct02_18-28-03/weight/stage1.pth"
NSGA_path="./NSGA/cifar10.csv"

python search_submodel.py \
  --stage $STAGE \
  --batch_size $BATCH_SIZE \
  --dataset $DATASET \
  --stage1_checkpoint_path $CHECKPOINT \
  --nsga_path $NSGA_path \
  --device $DEVICE \
`

stage 2 training:

`#!/bin/bash

STAGE='stage2'
BATCH_SIZE=128
DATASET="cifar10_full"
DEVICE="cuda:1"
MAX_LR=1e-5
MIN_LR=1e-7
CHECKPOINT="./experiments/logs_weight/stage_1/cifar10_full/Oct02_18-28-03/weight/stage1.pth"
NSGA_path="./NSGA/cifar10.csv"
GEN_ID=300

python train_stage2.py \
  --stage $STAGE \
  --batch_size $BATCH_SIZE \
  --max_lr $MAX_LR \
  --min_lr $MIN_LR \
  --dataset $DATASET \
  --stage1_checkpoint_path $CHECKPOINT \
  --nsga_path $NSGA_path \
  --gen_id $GEN_ID \
  --device $DEVICE \
`


Could you kindly check if there is any issue with the training of the model? thanks.
